src/main/java/com/codeborne/selenide/webdriver/FirefoxDriverFactory.java

```diff
@@ -49,31 +56,60 @@ FirefoxOptions createFirefoxOptions(Config config, Proxy proxy) {
     firefoxOptions.addPreference("network.proxy.allow_hijacking_localhost", true);
 
     firefoxOptions.merge(createCommonCapabilities(config, proxy));
-    firefoxOptions = transferFirefoxProfileFromSystemProperties(firefoxOptions);
+
+    FirefoxProfile profile = Optional.ofNullable(firefoxOptions.getProfile()).orElseGet(FirefoxProfile::new);
+    setupDownloadsFolder(config, profile);
+    transferFirefoxProfileFromSystemProperties(profile);
+    firefoxOptions.setProfile(profile);
 
     return firefoxOptions;
   }
 
-  private FirefoxOptions transferFirefoxProfileFromSystemProperties(FirefoxOptions currentFirefoxOptions) {
+  private void setupDownloadsFolder(Config config, FirefoxProfile profile) {
+    if (profile.getStringPreference("browser.download.dir", "").isEmpty()) {
+      profile.setPreference("browser.download.dir", new File(config.downloadsFolder()).getAbsolutePath());
+      profile.setPreference("browser.helperApps.neverAsk.saveToDisk", popularContentTypes());
+      profile.setPreference("pdfjs.disabled", true);  // disable the built-in viewer
+      profile.setPreference("browser.download.folderList", 2); // 0=Desktop, 1=Downloads, 2="reuse last location"
+    }
+  }
+
+  String popularContentTypes() {
+    try {
+      return String.join(";", IOUtils.readLines(getClass().getResourceAsStream("/content-types.properties"), UTF_8));
+    }
+    catch (IOException e) {
+      return "text/plain;text/csv;application/zip;application/pdf;application/octet-stream;" +
+        "application/msword;application/vnd.ms-excel;text/css;text/html";
+    }
+  }
+
+  private void transferFirefoxProfileFromSystemProperties(FirefoxProfile profile) {
     String prefix = "firefoxprofile.";
-    FirefoxProfile profile = Optional.ofNullable(currentFirefoxOptions.getProfile())
-      .orElseGet(FirefoxProfile::new);
+
     for (String key : System.getProperties().stringPropertyNames()) {
       if (key.startsWith(prefix)) {
         String capability = key.substring(prefix.length());
         String value = System.getProperties().getProperty(key);
         log.debug("Use {}={}", key, value);
-        if (value.equals("true") || value.equals("false")) {
-          profile.setPreference(capability, Boolean.valueOf(value));
+        if (isBoolean(value)) {
+          profile.setPreference(capability, parseBoolean(value));
         }
-        else if (value.matches("^-?\\d+$")) { //if integer
-          profile.setPreference(capability, Integer.parseInt(value));
+        else if (isInteger(value)) {
+          profile.setPreference(capability, parseInt(value));
         }
         else {
           profile.setPreference(capability, value);
         }
       }
     }
-    return currentFirefoxOptions.setProfile(profile);
+  }
+
+  private boolean isBoolean(String value) {
+    return "true".equals(value) || "false".equals(value);
```

Comment:
Suppose makes sense to convert value to lower case before comparing.

Reply:
Or even better - just use `String.equalsIgnoreCase` method